### PR TITLE
[ENH] Implemented logic to allow expansion of long dataset names

### LINKED ISSUE #165: [ENH] Results Expansion
### Is there an existing issue for this?

- [X] I have searched the existing issues

### New feature

I think there should be a mechanic for expanding the studies in the results so you can look at them in more detail. Only having part of the title and the flags that it matches is usually not enough to determine whether or not a study has what you're looking for. 

You could also include more information about the study in the initial results screen as well by including the entire title of the article, the author(s), and the doi. That way even if you don't have an option for expanding the article itself the user can search for it by looking up any combination of the given information.

### Unclear documentation

_No response_

### LINKED ISSUE #284: [ENH] Expand long dataset titles from ellipses
See also neurobagel/query-tool#68

### PR DESCRIPTION
**Pull request by rmanaem**
- Related to neurobagel/query-tool#68 
- Closes neurobagel/old-query-tool#284 

Changes proposed in this pull request:

- Added a `<span>` and a condition to truncate/expand the title of a card with a dataset name with more than 60 characters
- Added test for card title expansion

## Checklist

- [x] PR has an interpretable title with a prefix (`[ENH]`, `[FIX]`, `[REF]`, `[TST]`, `[CI]`, `[MNT]`, `[INF]`, `[MODEL]`, `[DOC]`) _(see https://neurobagel.org/contributing/pull_requests for more info)_
- [x] PR links to GitHub issue with mention `Closes #XXXX`
- [x] Tests pass
- [x] Checks pass

For new features:
- [x] Tests have been added

For bug fixes:
- [ ] There is at least one test that would fail under the original bug conditions.

### CODE REVIEW

File: cypress/component/ResultCard.cy.js [hunk 1]
```diff
@@ -33,4 +33,18 @@ describe('Result card', () => {
     cy.get('[data-cy="card-http://neurobagel.org/vocab/cool-dataset-checkbox"]').check();
     cy.get('@spy').should('have.been.calledWith', 'http://neurobagel.org/vocab/cool-dataset', true);
   });
+  it.only('Expands a long title and collapses it when clicked', () => {
+    props.datasetName = 'This is a veeeeeeeeeeeeeeeeeeeery long title that should be truncated';
```
Comment (surchs) [inline cypress/component/ResultCard.cy.js:37]: nice and clear, great stuff!

File: cypress/component/ResultCard.cy.js [hunk 2]
```diff
@@ -33,4 +33,18 @@ describe('Result card', () => {
     cy.get('[data-cy="card-http://neurobagel.org/vocab/cool-dataset-checkbox"]').check();
     cy.get('@spy').should('have.been.calledWith', 'http://neurobagel.org/vocab/cool-dataset', true);
   });
+  it.only('Expands a long title and collapses it when clicked', () => {
```
Comment (surchs) [inline cypress/component/ResultCard.cy.js]:
```suggestion
  it('Expands a long title and collapses it when clicked', () => {
```
from debug?

File: cypress/component/ResultCard.cy.js [hunk 3]
```diff
@@ -33,4 +33,18 @@ describe('Result card', () => {
...
+    props.datasetName = 'This is a veeeeeeeeeeeeeeeeeeeery long title that should be truncated';
+    cy.mount(ResultCard, {
+      listeners: {
+        'update-downloads': cy.spy().as('spy'),
+      },
+      propsData: props,
+    });
+    cy.get('[data-cy="card-http://neurobagel.org/vocab/cool-dataset-dataset"]').should('be.visible').contains('This is a veeeeeeeeeeeeeeeeeeeery long title that should be');
```
Comment (surchs) [inline cypress/component/ResultCard.cy.js]: I think a better assertion is `should.not.contain` "truncated"?

File: components/ResultCard.vue [hunk 4]
```diff
@@ -28,7 +28,13 @@
                     class="card-title"
                     :data-cy="`card-${datasetUuid}-dataset`"
                   >
-                    {{ datasetName }}
+                    {{ titleExpanded ? datasetName : datasetName.slice(0, 60) }}
+                    <span
+                      v-if="datasetName.length > 60"
```
Comment (surchs) [inline components/ResultCard.vue:33]: 60 linebreaks for me already, but that's obviously tied to screen size. Do you know when it truncated before? I think that would be a good starting point.

File: cypress/component/ResultCard.cy.js [hunk 5]
```diff
@@ -33,4 +33,18 @@ describe('Result card', () => {
     cy.get('[data-cy="card-http://neurobagel.org/vocab/cool-dataset-checkbox"]').check();
     cy.get('@spy').should('have.been.calledWith', 'http://neurobagel.org/vocab/cool-dataset', true);
   });
+  it.only('Expands a long title and collapses it when clicked', () => {
+    props.datasetName = 'This is a veeeeeeeeeeeeeeeeeeeery long title that should be truncated';
+    cy.mount(ResultCard, {
+      listeners: {
```
Comment (surchs) [inline cypress/component/ResultCard.cy.js]: do we need this listener guy here?

**Review by surchs — APPROVED** [review]
looks great, couple :cherries: comments, then :cook:

File: cypress/component/ResultCard.cy.js (hunk 2 above)
Comment (rmanaem) [reply cypress/component/ResultCard.cy.js]: Yea

File: cypress/component/ResultCard.cy.js (hunk 5 above)
Comment (rmanaem) [reply cypress/component/ResultCard.cy.js]: No, left over from copy pasta.

File: components/ResultCard.vue (hunk 4 above)
Comment (rmanaem) [reply components/ResultCard.vue:33]: I think it was 69 but taking into account the `...` and `more/less` I rounded down to 60.

**Pull request merged**